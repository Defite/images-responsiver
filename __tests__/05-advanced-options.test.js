'use strict';

const prettier = require('prettier');
const imagesResponsiver = require('../index');

function cleanHtml(html) {
  return prettier.format(html, { parser: 'html' });
}

describe('images with advanced options', () => {
  test('selector', () => {
    const content = `<!DOCTYPE html><body>
      <img src="test1.png" /><img src="test2.png" class="notransform" /></body>`;
    const transformed = imagesResponsiver(content, {
      default: {
        selector: 'img:not(.notransform)',
        fallbackWidth: 480,
      },
    });

    expect(cleanHtml(transformed)).toMatchSnapshot();
  });

  test('resizedImageUrl', () => {
    const content = `<!DOCTYPE html><body>
      <img src="test.png" width="420"></body>`;
    const transformed = imagesResponsiver(content, {
      default: {
        resizedImageUrl: (src, width) => `${src}?w=${width}`,
        fallbackWidth: 480,
      },
    });

    expect(cleanHtml(transformed)).toMatchSnapshot();
  });

  test('runBefore', () => {
    const content = `<!DOCTYPE html><body>
      <img src="test.png"></body>`;
    const transformed = imagesResponsiver(content, {
      default: {
        runBefore: (image, document) => {
          image.setAttribute('src', 'another-test.png');
        },
      },
    });

    expect(cleanHtml(transformed)).toMatchSnapshot();
  });

  test('runAfter', () => {
    const content = `<!DOCTYPE html><body>
      <img src="test.png"></body>`;
    const transformed = imagesResponsiver(content, {
      default: {
        runAfter: (image, document) => {
          image.dataset.after = 'hello!';
        },
      },
    });

    expect(cleanHtml(transformed)).toMatchSnapshot();
  });
});
