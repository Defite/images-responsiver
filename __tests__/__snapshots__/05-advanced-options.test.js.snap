// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`images with advanced options resizedImageUrl 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      src=\\"test.png?w=420\\"
      width=\\"420\\"
      srcset=\\"test.png?w=320 320w, test.png?w=420 420w\\"
      sizes=\\"100vw\\"
      data-pristine=\\"test.png\\"
    />
  </body>
</html>
"
`;

exports[`images with advanced options runAfter 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      src=\\"test-640.png\\"
      srcset=\\"
        test-320.png   320w,
        test-560.png   560w,
        test-800.png   800w,
        test-1040.png 1040w,
        test-1280.png 1280w
      \\"
      sizes=\\"100vw\\"
      data-pristine=\\"test.png\\"
      data-after=\\"hello!\\"
    />
  </body>
</html>
"
`;

exports[`images with advanced options runBefore 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      src=\\"another-test-640.png\\"
      srcset=\\"
        another-test-320.png   320w,
        another-test-560.png   560w,
        another-test-800.png   800w,
        another-test-1040.png 1040w,
        another-test-1280.png 1280w
      \\"
      sizes=\\"100vw\\"
      data-pristine=\\"another-test.png\\"
    />
  </body>
</html>
"
`;

exports[`images with advanced options selector 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      src=\\"test1-480.png\\"
      srcset=\\"
        test1-320.png   320w,
        test1-560.png   560w,
        test1-800.png   800w,
        test1-1040.png 1040w,
        test1-1280.png 1280w
      \\"
      sizes=\\"100vw\\"
      data-pristine=\\"test1.png\\"
    /><img src=\\"test2.png\\" class=\\"notransform\\" />
  </body>
</html>
"
`;
